tools/create_minimal_python.py

```diff
@@ -0,0 +1,73 @@
+from importlib import machinery
+import os
+import shutil
+import sys
+
+# These are modules which otherwise *should* be removed, but can't since they
+# are requirements of pytest.
+
+# TODO: Find a solution to this, possibly involving running tests in a
+# separate, more limited process.
+
+PYTEST_REQUIREMENTS = [
+    '_posixsubprocess', 'select', 'binascii', 'pyexpat', '_socket'
```

Comment:
I don't have a certain plan, other than to run the tests in a separate Python from py.test itself, which should be possible with pytest-xdist.  It's made even more complicated by the fact that matplotlib's own testing framework also needs subprocess (to run inkscape and ghostscript etc.)  Lord knows what the performance will be like.  I sort of see this as a first pass of "better than nothing for now."

Reply:
Again, how hard would it be for pyodide to just provide shims of subprocess and multiprocessing similar to https://github.com/matplotlib/matplotlib/blob/v2.2.2/lib/matplotlib/compat/subprocess.py (which we have recently removed)?
That'll make life easier for all downstream projects...

Reply:
What we be the purpose of this if it didn't actually work?  Just to get things to import?

Reply:
Yes.  It seems to be a bit excessive to ask each and every project to wrap their "import subprocess" (most of which are at the toplevel) in a try... except; at least if you provide shims then the discussion becomes "what features of the project don't work when subprocess isn't present" rather than "the thing doesn't even import".